### PR TITLE
Add more LIKE tests (null, missing, mistyped, variable type args)

### PR DESCRIPTION
**Pull request by alancai98**
Adds more tests for `LIKE` dealing with:
- non-literal arguments
- NULL + MISSING propagation
- mistyped arguments

Unknown propagation rules follow from section [8.5 of the SQL-92 spec](http://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt):

```
         1) If an <escape character> is specified and M, P, or E is the null
            value, then

              M LIKE P ESCAPE E

            is unknown.

         2) If an <escape character> is not specified and M or P is the null
            value, then

              M LIKE P

            is unknown.
```

NULL/MISSING and mistyped arguments follow [section 7 of the PartiQL spec](https://partiql.org/assets/PartiQL-Specification.pdf#section.7).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-tests-data/eval/primitives/operators/like.ion [hunk 1]
```diff
@@ -1569,6 +1569,32 @@ like::[
       ]
     }
   },
+  {
+    name:"like expression value, pattern, and escape as variable references",
+    statement:'''
+      SELECT v FROM
```
Comment (jpschorr) [inline partiql-tests-data/eval/primitives/operators/like.ion:1575]: since `v` is the same in two test cases, maybe change `v` in one of them to make it explicit which row is being filtered, or do something like below:

statement:
```
SELECT v LIKE p ESCAPE e as is_like FROM [
   {'v': 'abc%', 'p': 'abc/%', 'e': '/'},
   {'v': 'abcd', 'p': 'abc%', 'e': '/'},
   {'v': 'abcd', 'p': 'abc', 'e': '/'}
] AS tbl
   ```

result:
```
[
  {
    'is_like': true
  },
  {
    'is_like': true
  },
  {
    'is_like': false
  }
]
```

**Review by jpschorr — APPROVED** [review]

File: partiql-tests-data/eval/primitives/operators/like.ion [hunk 2]
```diff
@@ -1569,6 +1569,32 @@ like::[
       ]
     }
   },
+  {
```
Comment (am357) [inline partiql-tests-data/eval/primitives/operators/like.ion:1572]: I think, adding the same note as the PR's conversation can help understanding _why_ of these in the future.

**Review by am357 — DISMISSED** [review]

File: partiql-tests-data/eval/primitives/operators/like.ion (hunk 2 above)
Comment (alancai98) [reply partiql-tests-data/eval/primitives/operators/like.ion:1572]: Which note are you referring to?

Comment (alancai98) [reply partiql-tests-data/eval/primitives/operators/like.ion:1572]: Oh do you mean Josh's comment?

Comment (am357) [reply partiql-tests-data/eval/primitives/operators/like.ion:1572]: Oh no, I mean the description that you've put on the PR: https://github.com/partiql/partiql-tests/pull/49#issue-1560609110 itself.

Comment (alancai98) [reply partiql-tests-data/eval/primitives/operators/like.ion:1572]: Got it. I'll add a comment for that to the relevant section for null/missing propagation.

File: partiql-tests-data/eval/primitives/operators/like.ion (hunk 1 above)
Comment (alancai98) [reply partiql-tests-data/eval/primitives/operators/like.ion:1575]: Yeah, making it explicit will make the test clearer. I'll use your example statement.

**Review by am357 — APPROVED** [review]
